kernel/dw1000.c

```diff
@@ -1899,20 +1954,28 @@ static void dw1000_rx_irq(struct dw1000 *dw)
 		goto err_data;
 	}
 
-	/* Discard if an overrun occurred */
-	if (!len)
-		goto err_discard;
-
 	/* Hand off to IEEE 802.15.4 stack */
 	ieee802154_rx_irqsafe(dw->hw, skb, rx->lqi);
 	return;
 
- err_discard:
  err_data:
 	dev_kfree_skb_any(skb);
  err_alloc:
  err_info:
 	return;
+
+ err_recover:
+	/* Send recovery SPI message */
+	if (!DW1000_RX_IN_SYNC(status)) {
+		if ((rc = spi_sync(dw->spi, &rx->rcvr)) != 0) {
+			dev_err(dw->dev, "RX recovery#1 failed: %d\n", rc);
+		}
+	}
+
+	/* Send recovery SPI message */
+	if ((rc = spi_sync(dw->spi, &rx->rcvr)) != 0) {
+		dev_err(dw->dev, "RX recovery#2 failed: %d\n", rc);
+	}
```

Comment:
Note to self: double check that there is no possible path in which either the skb is leaked, or an uninitialised skb is used.